packages/kotti-ui/source/kotti-button/KtButton.vue

```diff
@@ -69,11 +71,29 @@ export default defineComponent<KottiButton.PropsInternal>({
 					'KtButton: Guideline Uncompliance: attempted to use helpText with a button that already has a label.',
 				)
 			}
+
+			if (
+				isToggle.value &&
+				props.type !== KottiButton.Type.DEFAULT &&
+				props.type !== KottiButton.Type.TEXT
+			) {
+				throw new Error(
+					'KtButton: Guideline Uncompliance: attempted to use toggleStatus with a button of type different than "DEFAULT" or "TEXT"',
+				)
+			}
 		})
 
 		return {
 			contentRef,
-			handleClick: (event) => emit('click', event),
+			handleClick: (event) =>
+				isToggle.value
+					? emit(
+							'update:toggleStatus',
+							props.toggleStatus === KottiButton.ToggleStatus.OFF
+								? KottiButton.ToggleStatus.ON
+								: KottiButton.ToggleStatus.OFF,
+					  )
+					: emit('click', event),
```

Comment:
@Juli0GT I wouldn't make this a ternary because we still want to emit the click in case the user app needs it